Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -356,18 +391,40 @@ open class NavigationMapView: UIView {
         
         mostRecentUserCourseViewLocation = location
         
-        // While animating to overview mode, don't animate the puck.
-        let duration: TimeInterval = animated && navigationCamera.state != .transitionToOverview ? 1 : 0
-        UIView.animate(withDuration: duration, delay: 0, options: [.curveLinear]) { [weak self] in
-            guard let screenCoordinate = self?.mapView.screenCoordinate(for: location.coordinate) else { return }
-            self?.userCourseView.center = CGPoint(x: screenCoordinate.x, y: screenCoordinate.y)
+        if customizedPuckType != nil {
```

Comment:
I do not fully understand this logic. In public API you expose `customizedPuckType` and based on it draw custom location puck. At the same time only `customizedPuck3DConfiguration` used. Also not sure why we need to stop updating location in Maps SDK location provider. Is it because of interference with location provided by NavigationNative?

I guess we should aim for such puck behavior:
- When only raw location is available we'll use it for the puck.
- If `PassiveLocationDataSource` is used we'll use snapped location for the puck.
- In case of active guidance puck will always use snapped location.

Reply:
For the `customizedPuckType`, it won't be `nil` if either `customizedPuck3DConfiguration ` or `customizedPuck2DConfiguration ` is used to draw puck.

The problem is that when under simulation mode, the `locationProvider` of the default `mapView.location` will continue sending snapped location update to the puck. If we want the puck only to receive location from the `SimulatedLocationManager`, we could only do it from here (because now we don't provide a wrapped locationProvider for the `SimulatedLocationManager`). And if we didn't stop the location update from the snapped location provider, it would show a puck shifting behavior. 

And if `PassiveLocationDataSource` is used, and used by `mapView.location.overrideLocationProvider(with:)`. The user puck will default receiving location update from the locationProvider, the `PassiveLocationDataSource`, not the snapped one, unless we override it with the `AppleLocationProvider()`.

So maybe we should check the simulation mode, if it's under simulation, it should only receive location update from the `SimulatedLocationManager` through the `public func updateUserCourseView(_ : animated:)` . Otherwise, we just need to allow the user puck receive the location from it's `locationProvider`.